Description: patching the build files to build on mips64el
Author: Pierre Gruet <pgt@debian.org>
Forwarded: no
Last-Update: 2024-07-06

--- a/make/build-jogl.xml
+++ b/make/build-jogl.xml
@@ -1464,6 +1464,12 @@
       <property name="linker.cfg.id.os"                    value="linker.cfg.linux.jogl.x11" />
     </target>
     
+    <target name="c.configure.linux.mips64el" if="isLinuxMips64el">
+      <echo message="Linux.MIPS64EL" />
+      <property name="compiler.cfg.id"                     value="compiler.cfg.linux" />
+      <property name="linker.cfg.id.os"                    value="linker.cfg.linux.jogl.x11" />
+    </target>
+
     <target name="c.configure.linux.ppc" if="isLinuxPpc">
       <echo message="Linux.PPC" />
       <property name="compiler.cfg.id"                     value="compiler.cfg.linux" />
@@ -1506,7 +1512,7 @@
       <property name="linker.cfg.id.os"                    value="linker.cfg.linux.jogl.x11" />
     </target>
 
-    <target name="c.configure.linux" depends="c.configure.linux.arm,c.configure.linux.aarch64,c.configure.linux.x86,c.configure.linux.alpha,c.configure.linux.amd64,c.configure.linux.ia64,c.configure.linux.hppa,c.configure.linux.mips,c.configure.linux.mipsel,c.configure.linux.ppc,c.configure.linux.ppc64,c.configure.linux.ppc64le,c.configure.linux.riscv64,c.configure.linux.s390,c.configure.linux.s390x,c.configure.linux.sparc,c.configure.x11" if="isLinux" />
+    <target name="c.configure.linux" depends="c.configure.linux.arm,c.configure.linux.aarch64,c.configure.linux.x86,c.configure.linux.alpha,c.configure.linux.amd64,c.configure.linux.ia64,c.configure.linux.hppa,c.configure.linux.mips,c.configure.linux.mipsel,c.configure.linux.mips64el,c.configure.linux.ppc,c.configure.linux.ppc64,c.configure.linux.ppc64le,c.configure.linux.riscv64,c.configure.linux.s390,c.configure.linux.s390x,c.configure.linux.sparc,c.configure.x11" if="isLinux" />
 
     <target name="c.configure.android" if="isAndroid">
       <echo message="Android" />
--- a/make/build-nativewindow.xml
+++ b/make/build-nativewindow.xml
@@ -744,6 +744,13 @@
       <property name="linker.cfg.id.drm"                    value="linker.cfg.linux.nativewindow.drm" />
     </target>
 
+    <target name="c.configure.linux.mips64el" if="isLinuxMips64el">
+      <echo message="Linux.MIPS64EL" />
+      <property name="compiler.cfg.id"                      value="compiler.cfg.linux" />
+      <property name="linker.cfg.id.oswin"                  value="linker.cfg.linux.nativewindow.x11" />
+      <property name="linker.cfg.id.drm"                    value="linker.cfg.linux.nativewindow.drm" />
+    </target>
+
     <target name="c.configure.linux.ppc" if="isLinuxPpc">
       <echo message="Linux.PPC" />
       <property name="compiler.cfg.id"                      value="compiler.cfg.linux" />
@@ -791,7 +798,7 @@
       <property name="linker.cfg.id.drm"                    value="linker.cfg.linux.nativewindow.drm" />
     </target>
 
-    <target name="c.configure.linux" depends="c.configure.linux.arm,c.configure.linux.aarch64,c.configure.linux.x86,c.configure.linux.alpha,c.configure.linux.amd64,c.configure.linux.ia64,c.configure.linux.hppa,c.configure.linux.mips,c.configure.linux.mipsel,c.configure.linux.ppc,c.configure.linux.ppc64,c.configure.linux.ppc64le,c.configure.linux.riscv64,c.configure.linux.s390,c.configure.linux.s390x,c.configure.linux.sparc,c.configure.x11" if="isLinux" />
+    <target name="c.configure.linux" depends="c.configure.linux.arm,c.configure.linux.aarch64,c.configure.linux.x86,c.configure.linux.alpha,c.configure.linux.amd64,c.configure.linux.ia64,c.configure.linux.hppa,c.configure.linux.mips,c.configure.linux.mipsel,c.configure.linux.mips64el,c.configure.linux.ppc,c.configure.linux.ppc64,c.configure.linux.ppc64le,c.configure.linux.riscv64,c.configure.linux.s390,c.configure.linux.s390x,c.configure.linux.sparc,c.configure.x11" if="isLinux" />
 
     <target name="c.configure.android" if="isAndroid">
       <echo message="Android" />
--- a/make/build-newt.xml
+++ b/make/build-newt.xml
@@ -797,6 +797,16 @@
       <property name="linker.cfg.id.drm"                    value="linker.cfg.linux.newt.drm"/>
     </target>
 
+    <target name="c.configure.linux.mips64el" if="isLinuxMips64el">
+      <echo message="Linux.mips64el" />
+      <property name="compiler.cfg.id"                      value="compiler.cfg.linux" />
+      <condition property="linker.cfg.id.oswin"             value="linker.cfg.linux.newt.x11"
+                                                            else="linker.cfg.linux">
+          <isset property="isX11" />
+      </condition>
+      <property name="linker.cfg.id.drm"                    value="linker.cfg.linux.newt.drm"/>
+    </target>
+
     <target name="c.configure.linux.ppc" if="isLinuxPpc">
       <echo message="Linux.ppc" />
       <property name="compiler.cfg.id"                      value="compiler.cfg.linux" />
@@ -874,7 +884,7 @@
       <property name="linker.cfg.id.drm"                    value="linker.cfg.linux.newt.drm"/>
     </target>
 
-    <target name="c.configure.linux" depends="c.configure.linux.x86,c.configure.linux.alpha,c.configure.linux.amd64,c.configure.linux.arm,c.configure.linux.aarch64,c.configure.linux.hppa,c.configure.linux.mips,c.configure.linux.mipsel,c.configure.linux.ppc,c.configure.linux.ppc64,c.configure.linux.ppc64le,c.configure.linux.riscv64,c.configure.linux.s390,c.configure.linux.s390x,c.configure.linux.sparc,c.configure.linux.ia64,c.configure.x11" if="isLinux"/>
+    <target name="c.configure.linux" depends="c.configure.linux.x86,c.configure.linux.alpha,c.configure.linux.amd64,c.configure.linux.arm,c.configure.linux.aarch64,c.configure.linux.hppa,c.configure.linux.mips,c.configure.linux.mipsel,c.configure.linux.mips64el,c.configure.linux.ppc,c.configure.linux.ppc64,c.configure.linux.ppc64le,c.configure.linux.riscv64,c.configure.linux.s390,c.configure.linux.s390x,c.configure.linux.sparc,c.configure.linux.ia64,c.configure.x11" if="isLinux"/>
 
     <target name="c.configure.android" if="isAndroid">
       <echo message="Android" />
